mmengine/hooks/checkpoint_hook.py

```diff
@@ -455,10 +461,24 @@ def _save_checkpoint_with_step(self, runner, step, meta):
         # from all ranks, but skip the following procedure.
         if not is_main_process():
             return
-
+        
         save_file = osp.join(runner.work_dir, 'last_checkpoint')
         with open(save_file, 'w') as f:
             f.write(self.last_ckpt)  # type: ignore
+        
```

Comment:
Hi @wanghao9610 , thanks for your contribution. The latest checkpoint path has been recorded in `save_file = osp.join(runner.work_dir, 'last_checkpoint')` so is it necessary to create a symlink?

Reply:
@zhouzaida I noticed that the "save_file" will save the absolute path of the latest checkpoint, if we move/change the work_dir to another path/name then rerun the exp, it will auto load the specified ckpt of the save_file, which is a absolute path and it will raise a FileNotFound Error. It is a good function to relatively symlink latest checkpoint to latest.pth, we can easily find the latest ckpt and don't need to specify any name (latest.pth vs iter_xxx.pth).  I also noticed that a more compatiable way, writing a relative path to the save_file ('last_checkpoint'). Summary, I suggest providing the two above functions is a useful practice. If you agree with me, I will make a new PR.

Reply:
At least, writing a relative path not a absolute path to save_file is nessary, the symlink of latest ckpt is optional.

Reply:
The reason for using the file `last_checkpoint` to save the latest checkpoint is that some checkpoints may be saved in cloud storage, and in such situations, it is not possible to create links or only record relative links.